src/Monolog/Handler/StreamHandler.php

```diff
@@ -99,6 +110,25 @@ protected function write(array $record)
 
     private function customErrorHandler($code, $msg)
     {
-        $this->errorMessage = preg_replace('{^fopen\(.*?\): }', '', $msg);
+        $this->errorMessage = preg_replace('{^(fopen|mkdir)\(.*?\): }', '', $msg);
+    }
+
+    /**
+     * @param string $stream
+     *
+     * @return null|string
+     */
+    private function getDirFromStream($stream)
+    {
+        $pos = strpos($stream, '://');
+        if ($pos === false) {
+            return dirname($stream);
+        }
+
+        if ('file:///' === substr($stream, 0, 8)) {
+            return DIRECTORY_SEPARATOR.dirname(substr($stream, 8));
```

Comment:
That DIRECTORY_SEPARATOR seems wrong to me on windows at least. IMO you should just check for `file://` and not with triple slash and then `return substr($stream, 7)` should do, because as it stands you break support for relative paths as far as I can see.